youtube_dl/extractor/soundcloud.py

```diff
@@ -160,13 +160,16 @@ def _extract_info_dict(self, info, full_title=None, quiet=False, secret_token=No
             'license': info.get('license'),
         }
         formats = []
+        query = {'client_id': self._CLIENT_ID}
+        if secret_token is not None:
+            query['secret_token'] = secret_token
         if info.get('downloadable', False):
             # We can build a direct link to the song
             format_url = (
-                'https://api.soundcloud.com/tracks/{0}/download?client_id={1}'.format(
-                    track_id, self._CLIENT_ID))
+                'https://api.soundcloud.com/tracks/{0}/download?{1}'.format(
+                    track_id, compat_urllib_parse_urlencode(query)))
```

Comment:
`update_url_query`.

Reply:
The code would be longer as it is using another format parameter. It's nothing more than more overhead here.

Reply:
It won't be longer.

Reply:
This is just an issue of preference. There are many other code written in this style, and on the other hand update_url_query has many redundant decode and encode.

Reply:
This the matter of **code reusage**, not personal preference. You should use the most generic solution available and not **reinvent the wheel**. Other code is this style was written before `update_url_query` was introduced.